go/state/state_db.go

```diff
@@ -827,6 +828,33 @@ func (s *stateDB) SetState(addr common.Address, key common.Key, value common.Val
 	}
 }
 
+func (s *stateDB) HasEmptyStorage(address common.Address) bool {
+	var (
+		isEmpty, shouldReturn bool
+	)
+	if !s.Exist(address) {
+		return true
+	}
+
+	s.data.ForEach(func(id slotId, val *slotValue) {
+		if id.addr == address && val.current != (common.Value{}) {
+			isEmpty = false
```

Comment:
The `isEmpty` flag is unnecessary. You initialize it to `false` and you update it here to `false`.

Simpler: rename "shouldReturn" to "hasNonZeroValue" and in line 847 just return `false`.